pkg/controller/serviceaccount/serviceaccounts_controller.go

```diff
@@ -197,8 +197,7 @@ func (c *ServiceAccountsController) syncNamespace(key string) error {
 	}
 
 	createFailures := []error{}
-	for i := range c.serviceAccountsToEnsure {
-		sa := c.serviceAccountsToEnsure[i]
```

Comment:
This isn't a typo. Allocating a clean sa var in the loop avoids mutating and passing a pointer to a shared variable around